internal/datacoord/compaction_trigger.go

```diff
@@ -625,46 +625,45 @@ func (t *compactionTrigger) generatePlans(segments []*SegmentInfo, force bool, i
 	var smallCandidates []*SegmentInfo
 	var nonPlannedSegments []*SegmentInfo
 
+	expectedSize := Params.DataCoordCfg.SegmentMaxSize.GetAsInt64() * 1024 * 1024
```

Comment:
We have a different segment maxSize for diskANN index `dataCoord.segment.diskSegmentMaxSize`

And we might not need to call `UpdateSegmentMaxSize()` in every trigger, since MaxRowNum is not needed here.